lib/index.js

```diff
@@ -1,46 +1,51 @@
 'use strict';
-var Assert = require('assert');
-var GoodSqueeze = require('good-squeeze');
-var Loggly = require('loggly');
-var Merge = require('lodash.merge');
 
-
-function GoodLoggly (events, config) {
-  if (!(this instanceof GoodLoggly)) {
-    return new GoodLoggly(events, config);
+const Assert = require('assert');
+const Stream = require('stream');
+const Loggly = require('loggly');
+const Merge = require('lodash.merge');
+
+
+class GoodLoggly extends Stream.Writable {
+  constructor (config) {
+    config = Merge({}, config);
+
+    Assert(typeof config.token === 'string', 'Loggly API token required');
+    Assert(typeof config.subdomain === 'string', 'Loggly subdomain required');
+
+    super({ objectMode: true, decodeStrings: false });
+    this._config = config;
+    this._client = Loggly.createClient({
+      token: config.token,
+      subdomain: config.subdomain,
+      json: true,
+      // Tags in JSON logs don't seem to appear properly in Loggly when
+      // sent with the X-LOGGLY-TAG header
+      useTagHeader: false,
+      tags: Array.isArray(config.tags) ? config.tags : []
+    });
   }
+  _write (event, encoding, callback) {
+    const data = Merge({}, event);
 
-  config = config || {};
-
-  Assert(!(events && events.ops), 'ops events are not supported by Loggly');
-  Assert(typeof config.token === 'string', 'Loggly API token required');
-  Assert(typeof config.subdomain === 'string', 'Loggly subdomain required');
+    // Map hapi event data fields to Loggly fields
+    if (this._config.name) {
+      data.name = this._config.name;
+    }
 
-  this._client = Loggly.createClient({
-    token: config.token,
-    subdomain: config.subdomain,
-    json: true,
+    if (this._config.hostname) {
+      data.hostname = this._config.hostname;
+    }
 
-    // Tags in JSON logs don't seem to appear properly in Loggly when sent with the X-LOGGLY-TAG header
-    useTagHeader: false,
-    tags: Array.isArray(config.tags) ? config.tags : []
-  });
-  this._config = Merge({}, config);
-  this._streams = {
-    squeeze: GoodSqueeze.Squeeze(events)
-  };
+    data.timestamp = GoodLoggly.timeString(data.timestamp);
```

Comment:
Yea. I thought so too. It was clearly done for improved testability. I think we should leave it. Same with `GoodLoggly.getMessage()`.